src/Catty/ViewController/Upload/UploadViewController.swift

```diff
@@ -259,171 +209,46 @@ class UploadViewController: UIViewController {
-                    }
-                    debugPrint("StatusCode is ", statusCode ?? "undefined")
-
-                    if statusCode == self.statusCodeOK {
-                        debugPrint("Upload successful")
-
-                        //Set unique Project-ID received from server
-                        if let aTag = dictionary?[self.projectIDTag], let project = self.project {
-                            let projectId = "\(aTag)"
-
-                            DispatchQueue.main.async(execute: {
-                                self.project?.rename(toProjectName: project.header.programName, andProjectId: projectId, andShowSaveNotification: true)
-                            })
-                        }
-
-                        //Set new token but when? everytime is wrong
-                        if let aTag = dictionary?[self.tokenParameterTag] {
-                            JNKeychain.saveValue(aTag, forKey: kUserLoginToken)
-                        }
-
-                        DispatchQueue.main.async(execute: {
-                            self.dismissView()
-                        })
-                    } else {
-                        let serverResponse = dictionary?[self.answerTag] as? String
-                        debugPrint("Error: ", serverResponse ?? "undefined")
-                        DispatchQueue.main.async(execute: {
-                            if serverResponse != nil {
-                                Util.alert(withText: serverResponse)
-                            } else {
-                                Util.alert(withText: kLocalizedUploadProblem)
-                            }
-                        })
-
-                        if statusCode == self.statusCodeTokenWrong {
-                            //Token not valid
-                            UserDefaults.standard.set(false, forKey: kUserIsLoggedIn)
-
-                            var viewArray: [AnyHashable]?
-                            if let aControllers = self.parent?.navigationController?.viewControllers {
-                                viewArray = aControllers
-                            }
-                            viewArray?.removeLast()
-                            var newViewArray: [AnyHashable]?
-                            if let anArray = viewArray {
-                                newViewArray = anArray
-                            }
-                            if let anArray = newViewArray as? [UIViewController] {
-                                self.parent?.navigationController?.setViewControllers(anArray, animated: true)
-                            }
-                        }
-                    }
-                }
+        if let project = project {
+            if let uploader = self.uploader {
+                uploader.upload(project: project,
+                                completion: { error in
+                                    self.enableUploadView()
+                                    if error == nil {
```

Comment:
The two nested if-else are not very readable. You could remove this and move to the else-statement of the` if let` statement. Or maybe better: return in the `if let `and use that (success) code for the default execution path.